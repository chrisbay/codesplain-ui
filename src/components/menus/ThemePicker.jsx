import React, { PropTypes } from 'react';
import {
  IconButton,
  IconMenu,
  MenuItem,
} from 'material-ui';
import memoize from 'lodash/memoize';
import ReactTooltip from 'react-tooltip';
import PaletteIcon from 'material-ui/svg-icons/image/palette';

import codeMirrorThemeOptions from '../../util/codemirror-theme-options';

const makeThemePickerItems = memoize(() => (
  codeMirrorThemeOptions.map(([displayName, value]) => (
    <MenuItem key={value} primaryText={displayName} value={value} />
  ))
));

const ThemePicker = (props) => {
  const { codeMirrorTheme, onChange } = props;
  return (
    <span data-tip data-for="theme-picker-tooltip">
      <IconMenu
        iconButtonElement={<IconButton><PaletteIcon /></IconButton>}
        onChange={onChange}
        value={codeMirrorTheme}
      >
        <MenuItem primaryText="Codesplain" value="codesplain" />
        {makeThemePickerItems()}
      </IconMenu>
      <ReactTooltip
        id="theme-picker-tooltip"
        effect="solid"
        place="bottom"
      >
        Select a theme
      </ReactTooltip>
    </span>
  );
};

ThemePicker.propTypes = {
  codeMirrorTheme: PropTypes.string,
  onChange: PropTypes.func.isRequired,
};

ThemePicker.defaultProps = {
  codeMirrorTheme: 'codesplain',
};

export default ThemePicker;
