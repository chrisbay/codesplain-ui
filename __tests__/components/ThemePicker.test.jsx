import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import getMuiTheme from 'material-ui/styles/getMuiTheme';

import ThemePicker from '../../src/components/menus/ThemePicker';

const defaultProps = {
  onChange: jest.fn(),
};

describe('<ThemePicker />', () => {
  const muiTheme = getMuiTheme();
  const shallowWithContext = node => shallow(node, { context: { muiTheme } });

  describe('snapshot tests', () => {
    it('matches default snapshot', () => {
      const wrapper = shallowWithContext(<ThemePicker {...defaultProps} />);
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
  });

  describe('prop: codeMirrorTheme', () => {
    it('is forwarded to underlying IconMenu', () => {
      const codeMirrorTheme = 'monokai';
      const wrapper = shallowWithContext(
        <ThemePicker
          {...defaultProps}
          codeMirrorTheme={codeMirrorTheme}
        />,
      );
      expect(wrapper.find('IconMenu').prop('value')).toEqual(codeMirrorTheme);
    });
  });

  describe('prop: onChange', () => {
    it('is forwarded to underlying IconMenu', () => {
      const onChange = () => {};
      const wrapper = shallowWithContext(
        <ThemePicker
          {...defaultProps}
          onChange={onChange}
        />,
      );
      expect(wrapper.find('IconMenu').prop('onChange')).toEqual(onChange);
    });
  });
});
