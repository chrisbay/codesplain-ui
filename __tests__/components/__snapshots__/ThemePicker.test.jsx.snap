exports[`<ThemePicker /> snapshot tests matches default snapshot 1`] = `
<span
  data-for="theme-picker-tooltip"
  data-tip={true}>
  <IconMenu
    anchorOrigin={
      Object {
        "horizontal": "left",
        "vertical": "top",
      }
    }
    animated={true}
    iconButtonElement={
      <IconButton
        disableTouchRipple={false}
        disabled={false}
        iconStyle={Object {}}
        tooltipPosition="bottom-center"
        touch={false}>
        <ImagePalette />
      </IconButton>
    }
    multiple={false}
    onChange={[Function]}
    onItemTouchTap={[Function]}
    onKeyboardFocus={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onRequestChange={[Function]}
    onTouchTap={[Function]}
    open={null}
    targetOrigin={
      Object {
        "horizontal": "left",
        "vertical": "top",
      }
    }
    touchTapCloseDelay={200}
    useLayerForClickAway={false}
    value="codesplain">
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Codesplain"
      value="codesplain" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="3024 Day"
      value="3024-day" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="3024 Night"
      value="3024-night" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="ABCDEF"
      value="abcdef" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Ambiance"
      value="ambiance" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Base 16 Dark"
      value="base16-dark" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Base 16 Light"
      value="base16-light" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Bespin"
      value="bespin" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Blackboard"
      value="blackboard" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Cobalt"
      value="cobalt" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Colorforth"
      value="colorforth" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Dracula"
      value="dracula" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Duotone Dark"
      value="duotone-dark" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Duotone Light"
      value="duotone-light" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Eclipse"
      value="eclipse" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Elegant"
      value="elegant" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Erlang Dark"
      value="erlang-dark" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Hopscotch"
      value="hopscotch" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Icecoder"
      value="icecoder" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Isotope"
      value="isotope" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Lesser Dark"
      value="lesser-dark" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Liquibyte"
      value="liquibyte" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Material"
      value="material" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Mbo"
      value="mbo" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="MDN-like"
      value="mdn-like" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Midnight"
      value="midnight" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Monokai"
      value="monokai" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Neat"
      value="neat" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Neo"
      value="neo" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Night"
      value="night" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Panda Syntax"
      value="panda-syntax" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Paraiso Dark"
      value="paraiso-dark" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Paraiso Light"
      value="paraiso-light" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Pastel On Dark"
      value="pastel-on-dark" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Railscasts"
      value="railscasts" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Rubyblue"
      value="rubyblue" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Seti"
      value="seti" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Solarized Dark"
      value="solarized-dark" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Solarized Light"
      value="solarized-light" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="The Matrix"
      value="the-matrix" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Tomorrow Night Bright"
      value="tomorrow-night-bright" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Tomorrow Night Eighties"
      value="tomorrow-night-eighties" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="TTCN"
      value="ttcn" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Twilight"
      value="twilight" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Vibrant Ink"
      value="vibrant-ink" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="XQ Dark"
      value="xq-dark" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="XQ Light"
      value="xq-light" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Yeti"
      value="yeti" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="Zenburn"
      value="zenburn" />
  </IconMenu>
  <ReactTooltip
    effect="solid"
    id="theme-picker-tooltip"
    insecure={true}
    place="bottom"
    resizeHide={true}
    wrapper="div">
    Select a theme
  </ReactTooltip>
</span>
`;
